src/compiler/checker.ts

```diff
@@ -1093,38 +1093,71 @@ namespace ts {
             return links.resolvedExports || (links.resolvedExports = getExportsForModule(moduleSymbol));
         }
 
-        function extendExportSymbols(target: SymbolTable, source: SymbolTable) {
+        interface ExportedSymbolDiagnostics {

```

Comment:
It's an structure used just to collect data to generate an error message from, so Diagnostics seems appropriate.